prow/tide/blockers/blockers.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package blockers
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/shurcooL/githubql"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+var (
+	branchRE = regexp.MustCompile(`(?im)\bbranch:[^\w-]*([\w-]+)\b`)
+)
+
+type githubClient interface {
+	Query(context.Context, interface{}, map[string]interface{}) error
+}
+
+// QUESTION: Do we need to worry about displaying Blockers for pools with no PRs?
+
+type Blocker struct {
+	Number int
+	URL    string
+	// TODO: time blocked? (when blocker label was added)
+}
+
+type Blockers struct {
+	// keyed by org/repo
```

Comment:
nit: could `type OrgRepo struct` and `type OrgRepoBranch struct` if we care about not stringly-typing the keys here

Reply:
Done.